pkgs/by-name/en/encore/go.nix

```
@@ -0,0 +1,80 @@
+{ fetchFromGitHub, stdenv, lib, go, ... }:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "go-encore";
+  version = "1.22.0";
+
+  src = fetchFromGitHub {
+    owner = "encoredev";
+    repo = "go";
+    rev = "encore-go${finalAttrs.version}";
+    hash = "sha256-SSa3CoUS6JMfs6T1PMb3NK8UeJnpLlTIn46/3oCTnL8=";
+  };
+
+  buildInputs = [ go ];
+
+  patchPhase = let
+    goSrc = fetchFromGitHub {
+      owner = "golang";
+      repo = "go";
+      rev = "release-branch.go${lib.versions.majorMinor finalAttrs.version}";
+      hash = "sha256-amASGhvBcW90dylwFRC2Uj4kOAOKCgWmFKhLnA9dOgg=";
+    };
+  in ''
+    # delete submodule and replace with the go source
```

I think it's good practice to ensure the pre- and post- hooks are run for each phase:
```suggestion
    runHook prePatch
    # delete submodule and replace with the go source
```